### PR TITLE
add blacklist of properties that can force layout/paint

### PR DESCRIPTION
**Pull request by erwinmombay**
couple of notes:
1. There might be a couple of things in here that were marked /*OK*/ but actually force layout/paint so it'll be good to review those.
2. these presubmit terms currently affects tests and comment (as well as private props) which is unfortunate, since our term searching code is pretty basic. I'll investigate in switching them to eslint rules since that might provide me more information about the runtime.

### CODE REVIEW

File: build-system/tasks/presubmit-checks.js [hunk 1]
```diff
@@ -38,6 +38,41 @@ var forbiddenTerms = {
   '\\.innerHTML': '',
   '\\.outerHTML': '',
   '\\.postMessage': '',
+  '\\.scrollTop': '',
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: done and done.

File: build-system/tasks/presubmit-checks.js [hunk 2]
```diff
@@ -38,6 +38,41 @@ var forbiddenTerms = {
...
+  '\\.clientHeight': '',
+  '\\.getClientRects': '',
+  '\\.getBoundingClientRect': '',
+  '\\.scrollBy': '',
+  '\\.scrollTo': '',
+  '\\.scrollIntoView': '',
+  '\\.scrollIntoViewIfNeeded': '',
+  '\\.scrollWidth': '',
+  '\\.scrollHeight': '',
+  '\\.scrollLeft': '',
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: done.

File: builtins/amp-ad.js [hunk 3]
```diff
@@ -192,7 +192,7 @@ export function installAd(win) {
      */
     getPlaceholderImage_() {
       let scores = scoreDimensions_(BACKFILL_DIMENSIONS_,
-          this.element.clientWidth, this.element.clientHeight);
+          this.element./*OK*/clientWidth, this.element./*OK*/clientHeight);
```
Comment (erwinmombay) [inline builtins/amp-ad.js]: done.

File: src/resources.js [hunk 4]
```diff
@@ -113,7 +113,7 @@ export class Resources {
     this.deferredMutates_ = [];
 
     /** @private {number} */
-    this.scrollHeight_ = 0;
+    this./*OK*/scrollHeight_ = 0;
```
Comment (erwinmombay) [inline src/resources.js]: done.

File: test/functional/test-resources.js [hunk 5]
```diff
@@ -297,9 +297,9 @@ describe('Resources discoverWork', () => {
   it('should re-render from requested position', () => {
     resource1.state_ = ResourceState_.LAYOUT_COMPLETE;
     resource2.state_ = ResourceState_.LAYOUT_COMPLETE;
-    resource1.element.getBoundingClientRect =
+    resource1.element./*OK*/getBoundingClientRect =
```
Comment (erwinmombay) [inline test/functional/test-resources.js]: done.

**Comment by erwinmombay** [comment]
closing this. i need to reopen a new PR with my new fork.

see new PR https://github.com/ampproject/amphtml/pull/505

File: build-system/tasks/presubmit-checks.js (hunk 1 above)
Comment (cramforce) [inline build-system/tasks/presubmit-checks.js]: Where?

**Pull request closed without merging**